ampd/src/starknet/verifier.rs

```diff
@@ -30,19 +33,83 @@ impl PartialEq<Message> for ContractCallEvent {
     }
 }
 
+pub fn verify_verifier_set(
+    event: &SignersRotatedEvent,
+    confirmation: &VerifierSetConfirmation,
+    source_gateway_address: &str,
+) -> Vote {
+    if event.signers.nonce != [0_u8; 32]
+        && event == confirmation
+        && event.from_address == source_gateway_address
+    {
+        Vote::SucceededOnChain
+    } else {
+        Vote::NotFound
+    }
+}
+
+impl PartialEq<VerifierSetConfirmation> for SignersRotatedEvent {
+    fn eq(&self, confirmation: &VerifierSetConfirmation) -> bool {
+        let expected = &confirmation.verifier_set;
+
+        // Convert and sort expected signers
+        let mut expected_signers = expected
+            .signers
+            .values()
+            .map(|signer| {
+                if let multisig::key::PublicKey::Ecdsa(pubkey) = &signer.pub_key {
+                    (pubkey.clone(), signer.weight.u128())
+                } else {
+                    // Skip non-ECDSA keys
+                    return (HexBinary::from_hex("").unwrap(), 0);
+                }
+            })
+            .collect::<Vec<_>>();
+        expected_signers.sort();
+
+        // Convert and sort actual signers from the event
+        let mut actual_signers = self
+            .signers
+            .signers
+            .iter()
+            .map(|signer| {
+                (
+                    HexBinary::from_hex(&signer.signer).unwrap(),
+                    signer.weight as u128,
+                )
+            })
+            .collect::<Vec<_>>();
+        actual_signers.sort();
+
+        // Compare signers, threshold, and created_at timestamp
+        actual_signers == expected_signers
```

Comment:
should signers_hash be compared?
I don't know if it even can be compared, but just asking